src/components/sdk/PreviewVideo/index.tsx

```diff
@@ -21,39 +22,42 @@ const StyledPreview = styled(VideoTile)`
 
 export const PreviewVideo: React.FC<BaseSdkProps> = (props) => {
   const audioVideo = useAudioVideo();
+  const meetingManager = useMeetingManager();
   const videoEl = useRef<HTMLVideoElement>(null);
-  const { selectedDevice } = useVideoInputs();
-  const { isVideoEnabled, setIsVideoEnabled } = useLocalVideo();
 
+  const { setIsVideoEnabled } = useLocalVideo();
+  // TODO: Move this to the Video Input Provider and expose only one selected Video Input device state
+  const [device, setDevice] = useState<Device | VideoTransformDevice | null>(meetingManager.selectedVideoInputTransformDevice);
+
+  // TODO: Move this to the Video Input Provider and expose only one selected Video Input device state
   useEffect(() => {
-    if (!audioVideo || !selectedDevice || !videoEl.current) {
-      return;
-    }
-    let mounted = true;
     const videoElement = videoEl.current;
+    meetingManager.subscribeToSelectedVideoInputTransformDevice(setDevice);
+    return () => {
+      if (videoElement) {
+        audioVideo?.stopVideoPreviewForVideoInput(videoElement);
+      }
+      meetingManager.unsubscribeFromSelectedVideoInputTranformDevice(setDevice);
+      setIsVideoEnabled(false);
+    };
+  }, []);
 
+  useEffect(() => {
+    if (!audioVideo || !device || !videoEl.current) {
+      return;
+    }
     async function startPreview() {
       if (!audioVideo) {
         return;
       }
-
-      await audioVideo.chooseVideoInputDevice(selectedDevice);
-      if (videoEl.current && mounted) {
+      await meetingManager.selectVideoInputDevice(device);
+      if (videoEl.current) {
```

Comment:
What is the reason for swapping `chooseVideoInputDevice` for `selectVideoInputDevice` in the implementation here?

Reply:
we want to rely on the meetingManager to wrap the sdk instead of calling the js sdk directly.